subsys/net/lib/wifi_credentials/wifi_credentials_shell.c

```diff
@@ -163,14 +163,23 @@ static int cmd_add_network(const struct shell *shell, size_t argc, char *argv[])
 			goto help;
 		}
 		creds.password_len = strlen(argv[3]);
-		if (
-			(creds.password_len > CONFIG_WIFI_CREDENTIALS_SAE_PASSWORD_LENGTH &&
-			 creds.header.type == WIFI_SECURITY_TYPE_SAE) ||
-			(creds.password_len > WIFI_PSK_MAX_LEN &&
-			 creds.header.type != WIFI_SECURITY_TYPE_SAE)
-		) {
-			shell_error(shell, "Password is too long for this security type");
-			goto help;
+		if (creds.header.type == WIFI_SECURITY_TYPE_SAE) {
+			if ((creds.password_len < WIFI_PSK_MIN_LEN) ||
+			    (creds.password_len > CONFIG_WIFI_CREDENTIALS_SAE_PASSWORD_LENGTH)) {
+				shell_error(shell,
```

Comment:
Typically we want to avoid this many levels of nested if's, because it makes the code harder to read.

Looks like the only difference for SAE is that it has a different max length. An alternative could be to check if type is SAE, and have a local variable/const that contains the max length. Another solution could be to leave the max length check as is, and add a separate check for min length. The latter would give the user a better error message in my opinion.

Reply:
Ok. Adding separate check for min length